lib/switchtec.c

```diff
@@ -1333,36 +1340,62 @@ static int log_c_to_file(struct switchtec_dev *dev, int sub_cmd_id, int fd)
 	return 0;
 }
 
+static int log_ram_flash_to_file(struct switchtec_dev *dev,
+				 int gen5_cmd, int gen4_cmd, int gen4_cmd_lgcy,
+				 int fd, FILE *log_def_file,
+				 struct switchtec_log_file_info *info)
+{
+	int ret;
+
+	if (switchtec_is_gen5(dev)) {
+		return log_a_to_file(dev, gen5_cmd, fd, log_def_file,
+				     info);
+	} else {
+		ret = log_a_to_file(dev, gen4_cmd, fd, log_def_file,
+				    info);
+
+		/* somehow hardware returns ERR_LOGC_PORT_ARDY_BIND
+		 * instead of ERR_SUBCMD_INVALID if this subcommand
+		 * is not supported, so we fall back to legacy
+		 * subcommand on ERR_LOGC_PORT_ARDY_BIND error */
+		if (ret > 0 &&
+		    ERRNO_MRPC(errno) == ERR_LOGC_PORT_ARDY_BIND)
```

Comment:
Maybe also check for ERR_SUBCMD_INVALID in case firmware fixes this bug?

Reply:
Good idea. Updated.